lib/matplotlib/dviread.py

```diff
@@ -861,19 +878,23 @@ def _register(self, words):
         # http://tex.stackexchange.com/questions/10826/
         # http://article.gmane.org/gmane.comp.tex.pdftex/4914
 
+        # input must be bytestrings (the file format is ASCII)
+        for word in words:
+            assert(isinstance(word, bytes))
+
```

Comment:
Again, the logic of this function may perhaps be more easily expressed using regexes.